<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BadPixelDetectorClass</class>
 <widget class="QMainWindow" name="BadPixelDetectorClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1320</width>
    <height>958</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BadPixelDetector</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,1,0">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>MIN</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="sliderDarkMin">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>10</number>
        </property>
        <property name="pageStep">
         <number>100</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>MAX</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="sliderDarkMax">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>10</number>
        </property>
        <property name="pageStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>15000</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="ChkShowPointsDark">
        <property name="text">
         <string>Show Bad Pixels</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="qyklabel" name="labelImageDark">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="mouseTracking">
         <bool>true</bool>
        </property>
        <property name="frameShape">
         <enum>QFrame::Box</enum>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Dark Image</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,0,0,0,0,0">
      <item>
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>MIN</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="sliderGainMin">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>10</number>
        </property>
        <property name="pageStep">
         <number>100</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>MAX</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="sliderGainMax">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>10</number>
        </property>
        <property name="pageStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>15000</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="ChkShowPointsGain">
        <property name="text">
         <string>Show Bad Pixels</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="qyklabel" name="labelImageGain">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="mouseTracking">
         <bool>true</bool>
        </property>
        <property name="frameShape">
         <enum>QFrame::Box</enum>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_7">
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Gain Image</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QPushButton" name="BtnDetectBadPixel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <italic>false</italic>
          <strikeout>false</strikeout>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Detect Bad Pixel</string>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <property name="autoRepeat">
         <bool>false</bool>
        </property>
        <property name="autoExclusive">
         <bool>false</bool>
        </property>
        <property name="autoDefault">
         <bool>false</bool>
        </property>
        <property name="default">
         <bool>false</bool>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="BtnAccumulatedDetect">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <italic>false</italic>
          <strikeout>false</strikeout>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Accumulated Detect</string>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <property name="autoRepeat">
         <bool>false</bool>
        </property>
        <property name="autoExclusive">
         <bool>false</bool>
        </property>
        <property name="autoDefault">
         <bool>false</bool>
        </property>
        <property name="default">
         <bool>false</bool>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Threshold(%)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEditPercentThre">
        <property name="text">
         <string>30</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Median size</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEditMedianSize">
        <property name="text">
         <string>5</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Bad line definition %</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEditBadLineDefPerc">
        <property name="text">
         <string>50</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="BtnManualAdd">
        <property name="text">
         <string>Add manual</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Row or pixel to mark</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEditManualAdd"/>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1320</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionLoadDarkImg"/>
    <addaction name="actionLoadGainImg"/>
    <addaction name="separator"/>
    <addaction name="actionSave_Cur_Dark_Image"/>
    <addaction name="actionSave_Cur_Gain_Image"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_Bad_Pixel_Map_2"/>
    <addaction name="actionSaveBadPixelMap"/>
   </widget>
   <widget class="QMenu" name="menuTool">
    <property name="title">
     <string>Tool</string>
    </property>
    <addaction name="actionDetectBadPixel"/>
    <addaction name="actionAccumulate_Bad_Pixel"/>
    <addaction name="actionUncorrectDarkImage"/>
    <addaction name="actionDo_replacement_Dark"/>
    <addaction name="actionDo_replacement_Gain"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTool"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionLoadDarkImg">
   <property name="text">
    <string>Load Dark Image</string>
   </property>
  </action>
  <action name="actionLoadGainImg">
   <property name="text">
    <string>Load Gain Image</string>
   </property>
  </action>
  <action name="actionSaveBadPixelMap">
   <property name="text">
    <string>Save Bad Pixel Map</string>
   </property>
  </action>
  <action name="actionDetectBadPixel">
   <property name="text">
    <string>Detect Bad Pixel(Re-detect)</string>
   </property>
  </action>
  <action name="actionUncorrectDarkImage">
   <property name="text">
    <string>UncorrectDarkImage</string>
   </property>
  </action>
  <action name="actionDo_replacement_Dark">
   <property name="text">
    <string>Do replacement_Dark</string>
   </property>
  </action>
  <action name="actionDo_replacement_Gain">
   <property name="text">
    <string>Do replacement_Gain</string>
   </property>
  </action>
  <action name="actionLoad_Bad_Pixel_Map">
   <property name="text">
    <string>Load Bad Pixel Map</string>
   </property>
  </action>
  <action name="actionLoad_Bad_Pixel_Map_2">
   <property name="text">
    <string>Load Bad Pixel Map</string>
   </property>
  </action>
  <action name="actionSave_Cur_Dark_Image">
   <property name="text">
    <string>Save Cur Dark Image</string>
   </property>
  </action>
  <action name="actionSave_Cur_Gain_Image">
   <property name="text">
    <string>Save Cur Gain Image</string>
   </property>
  </action>
  <action name="actionAccumulate_Bad_Pixel">
   <property name="text">
    <string>Accumulate Bad Pixels</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>qyklabel</class>
   <extends>QLabel</extends>
   <header>qyklabel.h</header>
   <slots>
    <slot>SetDrawPointToggle(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="crossremoval.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionLoadDarkImg</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_LoadDarkImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>506</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoadGainImg</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_LoadGainImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>506</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDetectBadPixel</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_DetectBadPixels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>506</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveBadPixelMap</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_SavePixelMap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>506</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionUncorrectDarkImage</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_UncorrectDark()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>506</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderDarkMin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_DrawDarkImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>10</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>905</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderDarkMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_DrawDarkImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>44</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>948</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderGainMin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_DrawGainImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>626</x>
     <y>957</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderGainMax</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_DrawGainImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>644</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>738</x>
     <y>957</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ChkShowPointsDark</sender>
   <signal>toggled(bool)</signal>
   <receiver>labelImageDark</receiver>
   <slot>SetDrawPointToggle(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>19</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>381</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ChkShowPointsGain</sender>
   <signal>toggled(bool)</signal>
   <receiver>labelImageGain</receiver>
   <slot>SetDrawPointToggle(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>593</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>985</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDo_replacement_Dark</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_DoReplacement_Dark()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>647</x>
     <y>506</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDo_replacement_Gain</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_DoReplacement_Gain()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>647</x>
     <y>506</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad_Bad_Pixel_Map_2</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_LoadBadPixelMap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>647</x>
     <y>506</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BtnDetectBadPixel</sender>
   <signal>clicked()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_DetectBadPixels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1250</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>1319</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_Cur_Dark_Image</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_SaveCurDark()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>647</x>
     <y>478</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_Cur_Gain_Image</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_SaveCurGain()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>647</x>
     <y>478</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAccumulate_Bad_Pixel</sender>
   <signal>triggered()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_AccumulateBadPixels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>647</x>
     <y>478</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BtnAccumulatedDetect</sender>
   <signal>clicked()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_AccumulateBadPixels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1271</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>1341</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BtnManualAdd</sender>
   <signal>clicked()</signal>
   <receiver>BadPixelDetectorClass</receiver>
   <slot>SLT_AddManual()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1241</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>659</x>
     <y>478</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>SLT_LoadDarkImage()</slot>
  <slot>SLT_LoadGainImage()</slot>
  <slot>SLT_DrawDarkImage()</slot>
  <slot>SLT_DrawGainImage()</slot>
  <slot>SLT_ShowBadPixels()</slot>
  <slot>SLT_DetectBadPixels()</slot>
  <slot>SLT_SavePixelMap()</slot>
  <slot>SLT_UncorrectDark()</slot>
  <slot>SLT_DoReplacement_Dark()</slot>
  <slot>SLT_DoReplacement_Gain()</slot>
  <slot>SLT_LoadBadPixelMap()</slot>
  <slot>SLT_SaveCurDark()</slot>
  <slot>SLT_SaveCurGain()</slot>
  <slot>SLT_AccumulateBadPixels()</slot>
  <slot>SLT_AddManual()</slot>
 </slots>
</ui>
